x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/embeddings/AzureOpenAiEmbeddingsServiceSettings.java

```diff
@@ -0,0 +1,339 @@
+        if (dimensionsSetByUser == null && map.containsKey(DIMENSIONS)) {
+            validationException.addValidationError(
+                ServiceUtils.missingSettingErrorMsg(DIMENSIONS_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+            );
+        }
+
+        Boolean encodingFormatSetByUser = removeAsType(map, ENCODING_FORMAT_SET_BY_USER, Boolean.class);
+        if (encodingFormatSetByUser == null && map.containsKey(ENCODING_FORMAT)) {
+            validationException.addValidationError(
+                ServiceUtils.missingSettingErrorMsg(ENCODING_FORMAT_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+            );
+        }
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new AzureOpenAiEmbeddingsServiceSettings(
+            settings,
+            Boolean.TRUE.equals(dimensionsSetByUser),
+            Boolean.TRUE.equals(encodingFormatSetByUser)
+        );
+    }
+
+    private static AzureOpenAiEmbeddingsServiceSettings fromRequestMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+
+        var commonFields = fromMap(map, validationException);
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new AzureOpenAiEmbeddingsServiceSettings(commonFields, commonFields.dimensions != null, commonFields.encodingFormat != null);
+    }
+
+    private static CommonFields fromMap(Map<String, Object> map, ValidationException validationException) {
+        String resourceName = extractRequiredString(map, RESOURCE_NAME, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        String deploymentId = extractRequiredString(map, DEPLOYMENT_ID, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        String apiVersion = extractRequiredString(map, API_VERSION, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        Integer dims = removeAsType(map, DIMENSIONS, Integer.class);
+        String encodingFormat = extractOptionalString(map, ENCODING_FORMAT, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        Integer maxTokens = removeAsType(map, MAX_INPUT_TOKENS, Integer.class);
+
+        if (dims != null) {
+            Boolean hasUserSetDimensions = extractOptionalBoolean(
+                map,
+                DIMENSIONS_SET_BY_USER,
+                ModelConfigurations.SERVICE_SETTINGS,
+                validationException
+            );
+
+            if (hasUserSetDimensions == null) {
+                validationException.addValidationError(
+                    ServiceUtils.missingSettingErrorMsg(DIMENSIONS_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+                );
+            }
+        }
+
+        if (encodingFormat != null) {
+            Boolean hasUserSetEncodingFormat = extractOptionalBoolean(
```

Comment:
The reason we need a `hasUserSetDimensions` is because we should only send dimensions in the request if the user explicitly set it. Since we programmatically set the dimensions field this was problematic because without having the field we wouldn't know if we set it or if the user did. If we just always sent it and the model being used was a v3 model, openai would return an error.

For the encoding format we have a couple options. We could always just send whatever the user specifies, since I believe it'll be accepted by azure regardless of the model version. If we parse this field, we'll want to default it to `float` because we need to expose it via the method

```
    public DenseVectorFieldMapper.ElementType elementType() {
        return <parsed or defaulted value here>;
    }
```

The other options is, since we only support float of the options that azure allows, maybe we don't parse the field right now and just always set it to `float`. If we go that route, I don't think we need to store the encoding formate in the persisted index.

In both cases, I don't think we need the `ENCODING_FORMAT_SET_BY_USER` field.

Reply:
> The other options is, since we only support float of the options that azure allows, maybe we don't parse the field right now and just always set it to float. If we go that route, I don't think we need to store the encoding formate in the persisted index.

I think this makes sense - especially given that the option for the encoding format is still in preview on Azure's side of things... I think the safe thing to do at the moment is to not expose the `encoding_format` to the user at all at the moment until there's GA for the option on Azure's side.